### PR TITLE
Make C/C++ project files configurable

### PR DESCRIPTION
**Pull request by proxict**
New global variable `g:ale_c_project_filenames` allows users to configure which files trigger a directory as a C/C++ project root.

Should address https://github.com/dense-analysis/ale/issues/2831

### CODE REVIEW

**Comment by proxict** [comment]
Can somebody review this please?

File: autoload/ale/c.vim [hunk 1]
```diff
@@ -192,7 +196,7 @@ function! ale#c#FindProjectRoot(buffer) abort
 
     " Fall back on detecting the project root based on other filenames.
     if empty(l:root)
-        for l:project_filename in g:__ale_c_project_filenames
+        for l:project_filename in g:ale_c_project_filenames
```
Comment (w0rp) [inline autoload/ale/c.vim:199]: The documentation says you can use `b:ale_c_project_filenames`, so make that true.

```suggestion
        for l:project_filename in ale#Var(a:buffer, 'c_project_filenames')
```

File: autoload/ale/c.vim [hunk 2]
```diff
@@ -5,8 +5,12 @@ call ale#Set('c_parse_makefile', 0)
 call ale#Set('c_parse_compile_commands', 0)
 let s:sep = has('win32') ? '\' : '/'
 
-" Set just so tests can override it.
-let g:__ale_c_project_filenames = ['.git/HEAD', 'configure', 'Makefile', 'CMakeLists.txt']
+let g:ale_c_project_filenames = get(g:, 'ale_c_project_filenames', [
+\   '.git/HEAD',
+\   'configure',
+\   'Makefile',
+\   'CMakeLists.txt',
```
Comment (w0rp) [inline autoload/ale/c.vim:12]:
```suggestion
\   'CMakeLists.txt',
\   'compile_flags.txt',
```

Let's add this here too, and write the same in the documentation for the option, so we do a little bit better project detection by default. You may have to break the default over many lines if it's long. You can use the hidden characters `>` and `<` in documentation to write multi-line code blocks. Search with `/>` and you'll find an example somewhere in the documentation.

**Review by w0rp — CHANGES_REQUESTED** [review]
This is a good idea.

This doesn't address #2831 totally. I would like to see another PR for parsing `compile_flags.txt`, which should be similar to parsing `command` keys from `compile_commands.json`. We could make our lives easy by reading the lines of output and converting them into the same JSON Dictionaries we get from `compile_commands.json`. We should always prefer `compile_commands.json` over `compile_flags.txt`, because the quoting of arguments will be better if the `arguments` keys are used in output, say from [Bear](https://github.com/rizsotto/Bear).

**Pull request closed without merging**